### ISSUE TITLE
Docker container cannot connect to WSL 2 Distro - no route to host

### ISSUE THREAD

**Issue by mdzhigarov** [docker/for-win#9168]
The issue kind of looks similar to the one reported here:
https://github.com/docker/for-win/issues/5167

Steps to reproduce:
1. Update to Docker Desktop 2.4.2.0
2. Enable WSL 2 Engine
3. docker build an image from provided Dockerfile
4. docker run -d a new container using the image
5. docker exec $CONTAINER ping $IP where $IP is the IP of the WSL 2 distro

`PING 172.17.90.237 (172.17.90.237) 56(84) bytes of data.
From 172.17.0.4 icmp_seq=1 Destination Host Unreachable
From 172.17.0.4 icmp_seq=2 Destination Host Unreachable
From 172.17.0.4 icmp_seq=3 Destination Host Unreachable`

Dockerfile
```
FROM  ubuntu:18.04
USER root
RUN apt update && apt install -y wget iptables iproute2 curl jq iputils-ping
CMD sleep 1000
```

Expected behavior:
The docker container should be able to reach the WSL2 host. This used to work fine before upgrade (unfortunately don't remember from which version I upgraded)

Actual behavior:
Ping from container to WSL2 distro fails with Destination Host Unreachable

**Comment by mdzhigarov** [comment]
The IP of the WSL Distro in the above example is 172.17.90.237, while Docker appears to be in 192.168...

![image](https://user-images.githubusercontent.com/1481379/96727784-1f9b8f80-13bc-11eb-900f-6f148dbb51f3.png)


Oddly enough, after restart my WSL 2 distro changed IP and now its IP is 192.168.83.234.
I can now successfully ping WSL 2 (192.168.83.234) from my container.

Any ideas what is causing this? It appears that it's not the update of Docker for Windows that's causing the issue but the fact that if WSL 2 Distro gets an IP in 172.17.* then it's not reachable from the docker containers

**Comment by mrsombre** [comment]
Same issue here.

```
Win 2004 19041
Docker Desktop 2.4.0.0
```

I started service inside WSL 2 Ubuntu on port 9003 and want to connect to it inside docker.

What I checked:

- WSL IP
```
5: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether 00:15:5d:93:30:ff brd ff:ff:ff:ff:ff:ff
    inet 172.19.87.209/20 brd 172.19.95.255 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 fe80::215:5dff:fe93:30ff/64 scope link
       valid_lft forever preferred_lft forever
```
- Connect from Win Host
```
PS C:\dev> curl 172.19.87.209:9003
test
```
- Connect from WSL itself
```
...@...:~/dev$ curl 172.19.87.209:9003
test
```
- Connect from inside container
```
$ docker-compose exec web bash
root@79381155354f:/var/www/app# curl 172.19.87.209:9003
curl: (7) Failed to connect to 172.19.87.209 port 9003: No route to host
```

TCP Port 9003 explicitly allowed for incoming in windows firewall in all environments.
Any suggestions?

**Comment by dewaffled** [comment]
I believe this is not specific to WSL: https://forums.docker.com/t/vpn-related-issues-docker-compose-on-ubuntu-18-04/92465

TLDR: Docker uses 172.17.0.0/16 subnet for internal network by default and external addresses from this subnet will not be accessible from the container.

This can be fixed by assigning any other unused subnet for internal network in `Docker Engine` settings:

```
{
  "bip":"192.168.200.1/24",
  ...
}
```
